### ISSUE TITLE
Terraform Warpgate Provider

### ISSUE THREAD

**Issue by andreee94** [warp-tech/warpgate#34]
This issue is a feature request.

Could be nice to have the possibility to automatically add new targets to warpgate via terraform directly when a new vm or LXC container is created.

This probably requires some api that the provider may call to add new targets.

A [datasource](https://www.terraform.io/language/data-sources) could be used to obtain the public key directly from the warpgate server to be injected to the `authorized_keys` file.

**Comment by stappersg** [comment]
The _terraform can add new targets_ is currently blocked by #24.

**Comment by stappersg** [comment]
For what it is worth:  Issue 24 has been completed.

**Comment by andreee94** [comment]
I started developing an unofficial provider at [https://github.com/andreee94/terraform-provider-warpgate](https://github.com/andreee94/terraform-provider-warpgate).
The documentaton is not perfect yet, but the provider should work (hopefully).

**Comment by Eugeny** [comment]
This is fantastic! Just a heads up - I'm not planning any drastic changes on the API side, but it's considered unstable until 1.0.0.  I've also seen your notice about `warpgate setup` - the latest image already includes `warpgate unattended-setup` for that very purpose ;)

**Comment by Eugeny** [comment]
https://registry.terraform.io/providers/warp-tech/warpgate/latest
